test/Npgsql.Tests/SystemTransactionTests.cs

```diff
@@ -341,7 +489,7 @@ public void SchemaConnectionShouldntEnlist()
         }
 
         void AssertNoPreparedTransactions()
-            => Assert.That(GetNumberOfPreparedTransactions(), Is.EqualTo(0));
+            => Assert.That(GetNumberOfPreparedTransactions(), Is.EqualTo(0), "Prepared transactions found");
 
         int GetNumberOfPreparedTransactions()
```

Comment:
> Do you mean that you were seeing 0 transactions in pg_prepared_xacts although you were expecting one to be there?

Yes.

> Do you consider this as evidence of the race condition you were mentioning before (or another race condition)?

No.

But maybe as an evidence that there is more differences between durable and volatile ressources. Enlisting many volatile ressources does not mandate promotion to distributed, provided there is no more than one durable ressource in the transaction.

> [As stated previously, a resource manager would make a volatile enlistment if it manages an in-memory, volatile resource. One of the benefits of using EnlistVolatile is that it does not force an unnecessary escalation of the transaction.](https://msdn.microsoft.com/en-us/library/ee818750.aspx)

> [As long as the System.Transactions infrastructure handles volatile resources and at most one durable resource that supports single-phase notifications, the transaction remains in the ownership of the System.Transactions infrastructure.](https://msdn.microsoft.com/en-us/library/ee818742.aspx)

The durable ressource used in the race test does not support single-phase, and with this one I see pg_prepared count raising. But for all the other tests, count stay 0.